src/main/java/core/basesyntax/service/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (Storage.fruits.containsKey(transaction.getFruit())
+                && transaction.getOperation() == FruitTransaction.Operation.PURCHASE) {
+            int newValue = Storage.fruits.get(transaction.getFruit())
+                    - transaction.getQuantity();
+            if (newValue < 0) {

```

Comment:
The error message 'Result can't be negative number' is not useful for debugging. It would be better to include more details, like the name of the fruit and the attempted quantity to purchase.